### ISSUE TITLE
False positive CA2012 with extension block

### ISSUE THREAD

**Issue by andryushchenko** [dotnet/roslyn-analyzers#7784]
### Describe the bug

False positive [CA2012](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2012) for ValueTask extensions when using extension block. No warnings for classic extension method or when calling as static method.

### To Reproduce

Simple code to reproduce with .NET SDK 10.0.101

```
await ValueTask.FromResult("test").Test1().ConfigureAwait(false); // OK
await ValueTask.FromResult("test").Test2().ConfigureAwait(false); // CA2012
await ExtensionTest.Test2(ValueTask.FromResult("test")).ConfigureAwait(false); //OK

public static class ExtensionTest
{
    public static async ValueTask Test1<T>(this ValueTask<T> valueTask)
    {
        Console.WriteLine(await valueTask.ConfigureAwait(false));
    }
    
    extension<T>(ValueTask<T> valueTask)
    {
        public async ValueTask Test2()
        {
            Console.WriteLine(await valueTask.ConfigureAwait(false));
        }
    }
}
```

### Expected behavior
No warnings
